### PR TITLE
Expand on hard and soft requirements

### PR DESCRIPTION
**Pull request by elharo**
@hboutemy

### CODE REVIEW

File: content/apt/pom.apt [hunk 1]
```diff
@@ -410,23 +410,30 @@ mvn install:install-file -Dfile=non-maven-proj.jar -DgroupId=some.group -Dartifa
...
 
-  * <<<[1.5,)>>>: x \>= 1.5
+  * <<<[1.0,2.0)>>>: 1.0 \<= x \< 2.0; Hard requirement for any version between 1.0 inclusive and 2.0 exclusive.
 
-  * <<<(,1.0],[1.2,)>>>: x \<= 1.0 or x \>= 1.2; multiple sets are comma-separated
+  * <<<[1.5,)>>>: Hard requirement for any version greater than or equal to 1.5.
 
-  * <<<(,1.1),(1.1,)>>>: this excludes 1.1 (for example if it is known not to work in combination with this library)
+  * <<<(,1.0],[1.2,)>>>: Hard requirement for any version less than or equal to 1.0 than or greater than
+    or equal to 1.2, but not 1.1. Multiple requirements are comma-separated
```
Comment (suztomo) [inline content/apt/pom.apt]:
```suggestion
    or equal to 1.2, but not 1.1. Multiple hard requirements are comma-separated.
```

File: content/apt/pom.apt [hunk 2]
```diff
@@ -410,23 +410,30 @@ mvn install:install-file -Dfile=non-maven-proj.jar -DgroupId=some.group -Dartifa
...
+
+  * <<<(,1.1),(1.1,)>>>: Hard requirement for any version except 1.1; for example because
+    it is known not to have a critical vulnerability.
```
Comment (suztomo) [inline content/apt/pom.apt]:
```suggestion
    the version is known to have a critical vulnerability.
```

File: content/apt/pom.apt [hunk 3]
```diff
@@ -410,23 +410,30 @@ mvn install:install-file -Dfile=non-maven-proj.jar -DgroupId=some.group -Dartifa
...
+  * <<<(,1.0]>>>: Hard requirement for any version \<= 1.0
 
-  * <<<[1.0,2.0)>>>: 1.0 \<= x \< 2.0
+  * <<<[1.2,1.3]>>>: Hard requirement for any version between 1.2 and 1.3 inclusive.
 
-  * <<<[1.5,)>>>: x \>= 1.5
+  * <<<[1.0,2.0)>>>: 1.0 \<= x \< 2.0; Hard requirement for any version between 1.0 inclusive and 2.0 exclusive.
 
-  * <<<(,1.0],[1.2,)>>>: x \<= 1.0 or x \>= 1.2; multiple sets are comma-separated
+  * <<<[1.5,)>>>: Hard requirement for any version greater than or equal to 1.5.
```
Comment (suztomo) [inline content/apt/pom.apt]:
```suggestion
  * <<<[1.5,)>>>: Hard requirement for any version \>= 1.5.
```

Consistency with line 424 `Hard requirement for any version \<= 1.0`

File: content/apt/pom.apt [hunk 4]
```diff
@@ -410,23 +410,30 @@ mvn install:install-file -Dfile=non-maven-proj.jar -DgroupId=some.group -Dartifa
...
+  If there are no versions of a dependency that satisfy all the hard requirements for that artifact, the build fails.
 
-  * <<<1.0>>>: "Soft" requirement on 1.0 (just a recommendation, if it matches all other ranges for the dependency)
+  Version requirements have the following syntax:
 
-  * <<<[1.0]>>>: "Hard" requirement on 1.0
+  * <<<1.0>>>: "Soft" requirement on 1.0. Use 1.0 if no other version appears earlier in the dependency tree.
 
-  * <<<(,1.0]>>>: x \<= 1.0
+  * <<<[1.0]>>>: "Hard" requirement for 1.0. Use 1.0 and only 1.0, even if other versions come before this dependency in
```
[the remark this answers is not in the document]
Comment (elharo) [reply content/apt/pom.apt]: removed

File: content/apt/pom.apt [hunk 5]
```diff
@@ -410,23 +410,30 @@ mvn install:install-file -Dfile=non-maven-proj.jar -DgroupId=some.group -Dartifa
...
+    the tree. If multiple hard versions conflict, fail the build.
```
[the remark this answers is not in the document]
Comment (elharo) [reply content/apt/pom.apt]: rewrote

File: content/apt/pom.apt (hunk 1 above)
Comment (elharo) [reply content/apt/pom.apt]: fixed

File: content/apt/pom.apt (hunk 2 above)
Comment (elharo) [reply content/apt/pom.apt]: fixed

**Comment by hboutemy** [comment]
thank you @elharo

**Pull request merged**